tests/Logging/LoggingTraitTest.php

```diff
@@ -0,0 +1,118 @@
+<?php
+/**
+ * Copyright 2024 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace Google\Auth\Tests\Logging;
+
+use Google\Auth\Logging\LoggingTrait;
+use Google\Auth\Logging\RpcLogEvent;
+use Google\Auth\Logging\StdOutLogger;
+use Google\Auth\Tests\BaseTest;
+use Psr\Log\LoggerInterface;
+
+class LoggingTraitTest extends BaseTest
+{
+    private MockClassWithLogger $loggerContainer;
+
+    public function setUp(): void
+    {
+        $this->loggerContainer = new MockClassWithLogger();
```

Comment:
nit: consider using anonymous classes because they're kinda cool

```suggestion
        $this->loggerContainer = new class() {
            use LoggingTrait;
            private LoggerInterface $logger;

            public function __construct()
            {
                $this->logger = new StdOutLogger();
            }

            public function logRequestEvent(RpcLogEvent $event): void
            {
                $this->logRequest($event);
            }

            public function logResponseEvent(RpcLogEvent $event): void
            {
                $this->logResponse($event);
            }
        };
```